langchain4j-azure-ai-search/src/main/java/dev/langchain4j/store/embedding/azure/search/AzureAiSearchEmbeddingStore.java

```diff
@@ -0,0 +1,426 @@
+                singletonList(id),
+                singletonList(embedding),
+                embedded == null ? null : singletonList(embedded));
+    }
+
+    private void addAllInternal(
+            List<String> ids, List<Embedding> embeddings, List<TextSegment> embedded) {
+        if (isNullOrEmpty(ids) || isNullOrEmpty(embeddings)) {
+            log.info("Empty embeddings - no ops");
+            return;
+        }
+        ensureTrue(ids.size() == embeddings.size(), "ids size is not equal to embeddings size");
+        ensureTrue(embedded == null || embeddings.size() == embedded.size(),
+                "embeddings size is not equal to embedded size");
+
+        List<Document> searchDocuments = new ArrayList<>();
+        for (int i = 0; i < ids.size(); ++i) {
+            Document document = new Document();
+            document.setId(ids.get(i));
+            document.setContent_vector(floatsArrayToList(embeddings.get(i).vector()));
+            if (embedded != null) {
+                document.setContent(embedded.get(i).text());
+                Document.Metadata metadata = new Document.Metadata();
+                metadata.setSource("langchain4j");
+                List<Document.Metadata.Attribute> attributes = new ArrayList<>();
+                for (Map.Entry<String, String> entry : embedded.get(i).metadata().asMap().entrySet()) {
+                    Document.Metadata.Attribute attribute = new Document.Metadata.Attribute();
+                    attribute.setKey(entry.getKey());
+                    attribute.setValue(entry.getValue());
+                    attributes.add(attribute);
+                }
+                metadata.setAttributes(attributes);
+                document.setMetadata(metadata);
+            } else {
+                document.setContent("");
+            }
+            searchDocuments.add(document);
+        }
+        List<IndexingResult> indexingResults = searchClient.uploadDocuments(searchDocuments).getResults();
+        for (IndexingResult indexingResult : indexingResults) {
+            if (!indexingResult.isSucceeded()) {
+                log.error("Failed to add embedding: {}", indexingResult.getErrorMessage());
+            } else {
+                log.info("Added embedding: {}", indexingResult.getKey());
+            }
+        }
+    }
+
+    private List<Float> floatsArrayToList(float[] floats) {
+        List<Float> list = new ArrayList<>();
+        for (float f : floats) {
+            list.add(f);
+        }
+        return list;
+    }
+
+    private float[] doublesListToFloatArray(List<Double> doubles) {
+        float[] array = new float[doubles.size()];
+        for (int i = 0; i < doubles.size(); ++i) {
+            array[i] = doubles.get(i).floatValue();
+        }
+        return array;
+    }
+
+    /**
+     * Calculates LangChain4J's RelevanceScore from Azure AI Search's score.
+     *
+     * Score in Azure AI Search is the cosine distance, and not the cosine similarity.
```

Comment:
seems that javadoc is not true: if score is cosine distance, then there is no need to do `double cosineDistance = (1 - score) / score;`